src/scmdata/ops.py

```diff
@@ -671,6 +672,175 @@ def delta_per_delta_time(self, out_var=None):
     return out
 
 
+def linear_regression(self):
+    """
+    Calculate linear regression of each timeseries
+
+    Note
+    ----
+    Times in seconds since 1970-01-01 are used as the x-axis for the
+    regressions. Such values can be accessed with
+    ``self.time_points.values.astype("datetime64[s]").astype("int")``. This
+    decision does not matter for the gradients, but is important for the
+    intercept values.
+
+    Returns
+    -------
+    list of dict[str : Any]
```

Comment:
Why not just return a DataFrame?

Reply:
Good question. 1) it makes life easier with the return gradient and return intercept options, 2) I want to keep the return values as pint types and encourage people to use them and putting pint types in dataframes always ends in tears in my experience if the unit can vary over the column. The the return gradient, return intercept and return scmrun options are fallbacks for those who don't want to deal with dictionaries.